test/dotcom/system_status/groups_test.exs

```diff
@@ -0,0 +1,401 @@
+defmodule Dotcom.SystemStatus.GroupsTest do
+  use ExUnit.Case, async: true
+
+  alias Dotcom.SystemStatus.Groups
+  alias Test.Support.Factories.Alerts.Alert
+  alias Test.Support.Factories.Alerts.InformedEntity
+
+  defp local_datetime(naive) do
+    DateTime.from_naive!(naive, "America/New_York")
+  end
+
+  defp now() do
+    local_datetime(~N[2025-01-09 13:00:00])
+  end
+
+  defp current_active_period() do
+    {local_datetime(~N[2025-01-09 10:00:00]), local_datetime(~N[2025-01-09 20:00:00])}
+  end
+
+  defp future_active_period(start_time) do
+    {local_datetime(start_time), local_datetime(~N[2025-01-09 20:00:00])}
+  end
+
+  defp past_active_period() do
+    {local_datetime(~N[2025-01-08 10:00:00]), local_datetime(~N[2025-01-08 20:00:00])}
+  end
+
+  defp statuses_for(groups, route_id) do
+    groups
+    |> Enum.filter(fn
+      %{route_id: ^route_id} -> true
+      %{} -> false
+    end)
+  end
+
+  describe "heavy rail groups" do
+    test "when there are no alerts, lists each line as normal" do
+      assert [
+               %{
+                 route_id: "Blue",
+                 sub_routes: [],
+                 statuses: [%{description: "Normal Service", time: nil}]
+               },
+               %{
+                 route_id: "Orange",
+                 sub_routes: [],
+                 statuses: [%{description: "Normal Service", time: nil}]
+               },
+               %{
+                 route_id: "Red",
+                 sub_routes: [],
+                 statuses: [%{description: "Normal Service", time: nil}]
+               },
+               %{
+                 route_id: "Green",
+                 sub_routes: [],
+                 statuses: [%{description: "Normal Service", time: nil}]
+               }
+             ] = Groups.groups([], now())
+    end
```

Comment:
The nesting, whitespace, and repetition hint at a cleaner way to do this.

Reply:
I want at least one test that validates not just the existence of those descriptions, but also the absence of other ones, plus at least one test that showcases the entire result struct. 

Basically, if we do a refactor that causes `Mattapan` to suddenly start showing up, or some other extra line that we don't actually want in the end-result, I want at least one test that will catch that.

I think I used the full data structure in two tests (copy-pasted three times), and then used a helper function to have the other tests assert only on the line that they care about.

Reply:
But, those are all separate tests and can be communicated more fully.